en/concept/architecture.md

```diff
@@ -1,36 +1,121 @@
 # Architectural Overview
 
-PX4 consists of two main layers: The [PX4 flight stack](../concept/flight_stack.md), an autopilot software solution and the [PX4 middleware](../concept/middleware.md), a general robotics middleware which can support any type of autonomous robot.
+PX4 consists of two main layers: the [flight stack](#flight-stack) is an autopilot software solution,
+and the [middleware](#middleware) is a general robotics layer that can support any type of autonomous robot.
```

Comment:
Again, this is a bit like saying "middleware is middleware". Perhaps "is a general robotics layer for internal/external communications and hardware integration". OK if you can't find something better.